### ISSUE TITLE
[Feature] Is it possible to make infinite scroll?

### ISSUE THREAD

**Issue by soft24** [bosskmk/pluto_grid#579]
If possible, is there any sample code?

I mean something like this:
https://flutter.syncfusion.com/#/datagrid/load-more-behavior/infinite-scrolling

**Comment by bosskmk** [comment]
@soft24 
It currently does not offer infinite scrolling.
You can implement it yourself, but I'll consider adding a feature.
thank you.

**Comment by SaltH2OFish** [comment]
> If possible, is there any sample code?
> 
> I mean something like this: https://flutter.syncfusion.com/#/datagrid/load-more-behavior/infinite-scrolling

I have implemented infinite scrolling in one of my projects by inserting the rows accordingly. Here is my insertRows routine.

`  void insertRows(int index, List<dynamic> items){
    Site item;
    List<PlutoRow> newRows=[];

    for (int i = 0; i < items.length; i++) {
      item = items[i];
      newRows.add(_createRow(item));
    }

    if (plutoGridStateManager!=null) plutoGridStateManager!.insertRows(index, newRows);
  }`

**Comment by bosskmk** [comment]
You can listen for an event when the end of the grid is reached as shown below.

```dart
stateManager.eventManager!.listener((PlutoGridEvent plutoEvent) {
  if (plutoEvent is PlutoGridCannotMoveCurrentCellEvent && plutoEvent.direction.isDown) {
    // do something like insertRows or stateManager.setShowLoading.
  }
}
```

`stateManager.scroll.bodyRowsVertical` also references `ScrollColtroller`. With this you can add rows same as above when scrolling reaches the end.

Of course, there are plans to provide it simply as a function of `PlutoGrid`.

**Comment by bosskmk** [comment]
Released PlutoGrid 5.3.0

Refer to below link.
https://github.com/bosskmk/pluto_grid/issues/601
